crates/hulk_behavior_simulator/src/simulator.rs

```diff
@@ -174,6 +179,81 @@ impl Simulator {
                 )?,
             )?;
 
+            self.lua.globals().set(
+                "get_robot_pose_x",
+                scope.create_function(|_lua, player_number: usize| {
+                    let player_number =
+                        to_player_number(player_number).map_err(LuaError::external)?;
+
+                    let position_x = self
+                        .state
+                        .lock()
+                        .robots
+                        .get_mut(&player_number)
+                        .unwrap()
+                        .database
+                        .main_outputs
+                        .ground_to_field
+                        .unwrap()
+                        .inner
+                        .translation
+                        .x;
+                    Ok(position_x)
+                })?,
+            )?;
+
+            self.lua.globals().set(
+                "get_robot_pose_y",
+                scope.create_function(|_lua, player_number: usize| {
+                    let player_number =
+                        to_player_number(player_number).map_err(LuaError::external)?;
+
+                    let position_y = self
+                        .state
+                        .lock()
+                        .robots
+                        .get_mut(&player_number)
+                        .unwrap()
+                        .database
+                        .main_outputs
+                        .ground_to_field
+                        .unwrap()
+                        .inner
+                        .translation
+                        .x;
+                    Ok(position_y)
+                })?,
+            )?;
+
+            self.lua.globals().set(
+                "whistle",
+                scope.create_function(|_lua, player_number: usize| {
+                    let player_number =
+                        to_player_number(player_number).map_err(LuaError::external)?;
+
+                    self.state
+                        .lock()
+                        .robots
+                        .get_mut(&player_number)
+                        .unwrap()
+                        .database
+                        .main_outputs
+                        .filtered_whistle
+                        .is_detected = true;
+                    let time = self.state.lock().time_elapsed;
+                    self.state
+                        .lock()
+                        .robots
+                        .get_mut(&player_number)
+                        .unwrap()
+                        .database
+                        .main_outputs
+                        .filtered_whistle
+                        .last_detection = Some(UNIX_EPOCH + time);
```

Comment:
Why + UNIX_EPOCH?

Reply:
Because this is the way the robots store time in the simulator. Time is just a duration of the simulator since start, UNIX_EPOCH makes it a "proper" system time.